### PR TITLE
Working brach for wallpaper engine.

### PR DESCRIPTION
**Pull request by mishaglik**
Mostly sharing ideas. Merging is not asked by now/

### CODE REVIEW

File: src/Engine/WallpaperEngine/Widget.hpp
Comment (Execorn) [inline src/Engine/WallpaperEngine/Widget.hpp:1]: Please, consider renaming IWidget to WidgetBase or WidgetImpl to improve code readability.

File: src/Engine/WallpaperEngine/MetaObject.hpp
Comment (Execorn) [inline src/Engine/WallpaperEngine/MetaObject.hpp:1]: Suggestion: add namespace 'rebind' for meta and meta_cast functions.

File: src/Engine/WallpaperEngine/Widget.hpp
Comment (mishaglik) [reply src/Engine/WallpaperEngine/Widget.hpp:1]: Done

**Comment by mishaglik** [comment]
So base ideas estabilished may try to merge

File: src/Engine/WallpaperEngine/Widget.hpp
Comment (Execorn) [inline src/Engine/WallpaperEngine/Widget.hpp:1]: Consider renaming m_drawOn to m_display_num or m_display_idx

Comment (mishaglik) [reply src/Engine/WallpaperEngine/Widget.hpp:1]: Done

**Pull request merged**